owltorrent-network/src/main/java/edu/rice/owltorrent/network/TorrentManager.java

```diff
@@ -357,6 +376,17 @@ public void reportBlockCompletion(FileBlockInfo blockInfo) {
         uncompletedPieces.put(pieceIndex, status);
       } else {
         completedPieces.add(pieceIndex);
+        // Advertise with Have message for more bandwidth
+        for (Peer leecher : leechers) {
+          if (peers.containsKey(leecher) && !leecher.isAmChoked() && leecher.isPeerInterested()) {
```

Comment:
I'm not sure, but do we/should we advertise to peers choking us for them to unchoke us?